lib/library.rb

```diff
@@ -0,0 +1,46 @@
+require 'book'
+require 'date'
+
+class Library
+   attr_accessor :books   #TODO duedate person status
+
+   def initialize(library)
+    @books = []
+   end
+
+   def check_in(book)
+    add = {item: book, status: :okay, person: :non, duedate: " " }
+    @books << add
+   end
+
+   def lend(title)
+    case
+     when books.detect{|a| a[:title] == title}[:status] = :lended
+       then 'Sorry, book is already with another reader'
+    when check = portfolio.select{|a| a[:duedate] > Date.today.strftime('%F')}
+      if check != []
+       then 'Sorry, you have one overdue book to return'
+    else
+        items = books.detect{|a| a[:title] == title}
+        items[:status] = :lended
+        items[:person] = :MEEE
+        items[:dudate] = duedate
+
+        #TODO: command to replace the original hash / or delete it
+    end
+  end
+
+ def return(title)
```

Comment:
Lint/NestedMethodDefinition: Method definitions must not be nested. Use `lambda` instead. (https://github.com/bbatsov/ruby-style-guide#no-nested-methods)